src/core/components/ChatHistory.tsx

```diff
@@ -158,18 +159,29 @@ const HistoryItem = memo(
       const trimmedTitle = newTitle.trim();
       if (trimmedTitle === '') {
         setNewTitle(title);
-      } else {
-        saveToLocalStorage(trimmedTitle);
-        setNewTitle(trimmedTitle);
```

Comment:
This was the root of the issue. Updating local storage is synchronous, so if we do this at the same time as updating the UI, we can see a significant lag for users with long histories.

https://github.com/user-attachments/assets/ada77df1-016c-4b8b-88cb-f8dbd530bbc8